src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -66,17 +75,32 @@ public override SdkResult Resolve(SdkReference sdkReference, SdkResolverContext
             List<string> warnings = null;
             CachingWorkloadResolver workloadResolver = null;
 
+            ResolverLogger logger = null;
+            if (_shouldLog)
+            {
+                logger = new ResolverLogger();
+            }
+
+            logger?.LogMessage($"Attempting to resolve MSBuild SDK {sdkReference.Name}");
+
             if (context.State is CachedState priorResult)
             {
+                logger?.LogString("Using previously cached state");
+
                 dotnetRoot = priorResult.DotnetRoot;
                 msbuildSdksDir = priorResult.MSBuildSdksDir;
                 netcoreSdkVersion = priorResult.NETCoreSdkVersion;
                 propertiesToAdd = priorResult.PropertiesToAdd;
                 workloadResolver = priorResult.WorkloadResolver;
+
+                logger?.LogMessage($"\tDotnet root: {dotnetRoot}");
```

Comment:
I think you mentioned perf concerns at one point. Would it be cheaper to log once and just do a comma separate list of values on a single line, e.g.

`logger?.LogMessage($"\tDotnet root: {dotnetRoot}, MSBuild SDKs Dir: {msbuildSdksDir}");`

Or was the perf concerns related to something else?

Reply:
The resolver is called a lot of times, so any increased allocation along the success path could impact VS and MSBuild perf.  I don't think there will be much difference between logging FormattableStrings compared to logging just one though.  After this ships I hope to do a memory comparison between when the feature is on versus off, and at that point we can combine these if need be.